Add section on the modal system (#39)
denniskigen/o3-docs: pages/docs/modal-system.en-US.mdx

import { Callout } from "nextra-theme-docs";

# The modal system

The modal system is a part of the framework that handles rendering modal dialogs, warnings, and other similar displays. Its main goal is to coordinate modals across different applications so that only a single modal is rendered for the user at a time without any other modals being removed.

> To use the modal system, first your application must register the modal, as described below. Once the modal is registered, call [`showModal`](https://o3-docs.openmrs.org/docs/framework-api-reference#showmodal) with the modal name and the framework will render the modal for you.

Reviewer: ```suggestion
To use the modal system, your application must register the modal as described below. Once the modal is registered, call [`showModal`](/docs/framework-api-reference#showmodal) with the modal name and the framework will render the modal for you.
```

import { Callout } from "nextra-theme-docs";

# The modal system

> An easier way to trigger a modal without having to have a conditional state to render it in the DOM.

Reviewer 2: ```suggestion
The modal system is a part of the framework that handles rendering modal dialogs, warnings, and other similar displays. Its main goal is to coordinate modals across different applications so that only a single modal is rendered for the user at a time without any other modals being removed.
```

> This is handled by the function [`showModal`](https://o3-docs.openmrs.org/docs/framework-api-reference#showmodal) defined in `esm-core` that takes care of the rendering of the modal.

Reviewer 2: ```suggestion
The main function users of the modal system use is the [`showModal`](https://o3-docs.openmrs.org/docs/framework-api-reference#showmodal) function defined in `esm-core`, which will render the named modal.
```

Author: > The main function users of the modal system use is the [`showModal`](https://o3-docs.openmrs.org/docs/framework-api-reference#showmodal) function

Sorry, Ian. I don't quite understand what this means. Could you please explain what you meant?

Reviewer 2: Maybe @denniskigen has a better suggeestion?

Author: I think the way you suggested instead of the "An easier way..." is perfect! 
I was referring to the suggestion below that, specifically the:
![image](https://github.com/denniskigen/o3-docs/assets/34070216/15583b28-e029-4b50-9766-a4a3b20c9e45)

Reviewer 2: Yeah, I got that! Maybe something like:

```suggestion
To use the modal system, first your application must register the modal, as described below. Once the modal is registered, call [`showModal`](https://o3-docs.openmrs.org/docs/framework-api-reference#showmodal) with the modal name and the framework will render the modal for you.
```

Author: Aaah gotcha! Sorry my bad. Thanks, Ian!

<Callout emoji="ℹ️" type="info">
  For more information about how `showModal` works, check out its [implementation](https://github.com/openmrs/openmrs-esm-core/blob/01c32c2090a47a881d5f9fa39650ec4a9037aa36/packages/framework/esm-styleguide/src/modals/index.tsx).
  The API reference can be found [here](/docs/framework-api-reference#showmodal).
</Callout>

## Usage

### Implementation of the modal

> Make a seperate component for the modal, as you normally would, but instead of the top most wrapper component being the `ComposedModal`, it should be a `React.Fragment`.

Reviewer 2: I'm not really sure I understand this line.

```suggestion
Make a seperate component for the modal, as you normally would, but instead of the top most wrapper component being the `ComposedModal`, it should be a `React.Fragment`.
```

Author: Sorry, I was trying to say that a `React.Fragment` should be at the top most of the component and not `ComposedModal`
